database/models/User.js

```diff
@@ -0,0 +1,55 @@
+const { DataTypes, Model } = require("sequelize")
+
+module.exports = (sequelize) => {
+  class User extends Model {
+    static associate(models) {
+      const { Site, SiteMember } = models
+      User.belongsToMany(Site, {
+        through: SiteMember,
+        onUpdate: "CASCADE",
+        onDelete: "CASCADE",
+      })
+    }
+  }
+
+  User.init(
+    {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: DataTypes.BIGINT,
+      },
+      email: {
+        allowNull: true,
+        unique: true,
+        type: DataTypes.TEXT,
+        validate: {
+          isEmail: true,
+        },
+      },
+      githubId: {
+        allowNull: false,
+        unique: true,
+        type: DataTypes.TEXT,
+        validate: {
+          notEmpty: true,
+        },
+      },
```

Comment:
Should we be allowing users with valid githubId but no email?

Reply:
Should we be putting email and githubId in separate tables for easier retrieval?